keymanapp/keyman
# feat[android]: Show progress to wait for catalog update

Fixes #2265

## Previous Behavior ##
Since catalog download is implemented as background job, it is possible to go to "Add language" and "Add Keyboard" dialog before catalog update is finished. It could happen, that the user gets an empty or only partial language/keyboard list.

## New Behavior ##
When opening the "Add language" and "Add Keyboard" dialog and a catalog update is in progress, a infinite non modal spinner dialog is shown.
User can go back and return later or wait until update is finished.

Goto Languages Settings
![Screenshot_1573195298](https://user-images.githubusercontent.com/52272980/68461877-e49d5c00-023d-11ea-9bd1-1208d5139449.png)

Try to add a language when catalog update is not finished:
![Screenshot_1573197274](https://user-images.githubusercontent.com/52272980/68461919-fed73a00-023d-11ea-88b9-d33c4cba599f.png)

When catalog is up to date:
![Screenshot_1573195377](https://user-images.githubusercontent.com/52272980/68461949-0bf42900-023e-11ea-9f4b-cc13b3ca3eed.png)

## Work Done ##
- Add new generic list_activity_with_progress
- Add hide and show method to switch between listview and progressview
- Change "Add Keyboard" and "Add Language" dialog

## Next Steps ##
- Start update catalog on startup, that the link on Getting started activity is working as well

## Testing 
- Reinstall keyman
- Goto Settings/ Installed languages
- Start download catalog notification is shown
- Click Add a new language
- Infinite progress is shown
- When download is finished, the language list is shown
